presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1874,7 +1873,12 @@ private HiveInsertTableHandle beginInsertInternal(ConnectorSession session, Conn
         Table table = metastore.getTable(metastoreContext, tableName.getSchemaName(), tableName.getTableName())
                 .orElseThrow(() -> new TableNotFoundException(tableName));
 
-        checkTableIsWritable(table, writesToNonManagedTablesEnabled, metastore.getTableConstraints(metastoreContext, tableName.getSchemaName(), tableName.getTableName()));
+        tableWritabilityChecker.checkTableWritable(table);
+
+        List<TableConstraint<String>> constraints = metastore.getTableConstraints(metastoreContext, tableName.getSchemaName(), tableName.getTableName());
+        if (constraints.stream().anyMatch(TableConstraint::isEnforced)) {
+            throw new PrestoException(NOT_SUPPORTED, format("Cannot write to table %s since it has table constraints that are enforced", table.getSchemaTableName().toString()));
+        }
```

Comment:
Why not move this as well into the checkTableWritable ?

Reply:
It feels like it's more of a Presto current limitation and should always be rejected (unless constraint checks are implemented). Or do you think there's a possibility that a connector derived from Hive can somehow allow such inserts?

Reply:
similar to the previous util method, i was hoping this new interface implementation to take care of all checks related to checking if a table is writable or not. I dont have a strong opinion here.

Reply:
Also it felt like it might not be very obvious to somebody implementing the interface of what these constraints really are. I don't really have a strong opinion here either. Maybe we can keep it out of the interface for now, and then once we see that an alternative implementation is needed (for the constraint checks) we can move them as a parameter to the interface?